specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-03-31-preview/datamigration.json

```diff
@@ -0,0 +1,1239 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Azure Data Migration Service Resource Provider",
+    "version": "2018-03-31-preview",
+    "description": "The Data Migration Service helps people migrate their data from on-premise database servers to Azure, or from older database software to newer software. The service manages one or more workers that are joined to a customer's virtual network, which is assumed to provide connectivity to their databases. To avoid frequent updates to the resource provider, data migration tasks are implemented by the resource provider in a generic way as task resources, each of which has a task type (which identifies the type of work to run), input, and output. The client is responsible for providing appropriate task type and inputs, which will be passed through unexamined to the machines that implement the functionality, and for understanding the output, which is passed back unexamined to the client.",
+    "x-ms-code-generation-settings": {
+      "name": "DataMigrationServiceClient",
+      "useDateTimeOffset": true
+    }
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.DataMigration/skus": {
+      "get": {
+        "summary": "Get supported SKUs",
+        "description": "The skus action returns the list of SKUs that DMS supports.",
+        "tags": [
+          "Standard operation",
+          "GET"
+        ],
+        "operationId": "ResourceSkus_ListSkus",
```

Comment:
I would change the name to ResourceSkus_List since Skus is already in the operation group, else you'd end up with a method like this.
```
ResourceSkus.ListSkus()
```

Reply:
@jhendrixMSFT  - If you look only at 2nd commit, you'll see the diff of what is new in this api-version. The previous version is already in public preview. This wasn't really changed in this version, it has been present even in the previous version.

Yes, the APIs have been reviewed and deployed.

Reply:
Yes, this can be published.
I've been updating the azure-sdk-for-net repo after this (for previous changes), not sure about the Go part.